src/components/card/card.tsx

```diff
@@ -1,98 +1,120 @@
-import { ClassNames } from '@/const'
+import { OfferCardClassNames, OFFER_URL } from '@/const'
 import cn from 'classnames'
 import FavoriteButton from '@/components/favorite-button/favorite-button'
+import { Link } from 'react-router-dom'
 
 type CardProps = {
+  id: string
   image: string
   price: number
   name: string
   rating: number
   width: string
   height: string
   type: string
-  className: ClassNames
+  className: OfferCardClassNames
   isPremium?: boolean
-  onMouseEnter?: () => void
-  onMouseLeave?: () => void
+  onCardAction?: (id: string | null) => void
   view: 'list' | 'favorites' | 'near'
 }
 
-const Card = (props: CardProps): JSX.Element => {
+const Card = ({
+  id,
+  image,
+  price,
+  name,
+  rating,
+  width,
+  height,
+  type,
+  className,
+  isPremium,
+  onCardAction,
+  view,
+}: CardProps): JSX.Element => {
+  const handleOnMouseEnter = () => {
+    onCardAction?.(id)
+  }
+
+  const handleOnMouseLeave = () => {
+    onCardAction?.(null)
+  }
+
   return (
     <article
-      className={`${cn(props.className, 'place-card')}`}
-      onMouseEnter={props.onMouseEnter}
-      onMouseLeave={props.onMouseLeave}
+      className={`${cn(className, 'place-card')}`}
+      onMouseEnter={handleOnMouseEnter}
+      onMouseLeave={handleOnMouseLeave}
     >
-      {props.view === 'list' && (
+      {view === 'list' && (
         <div className="cities__image-wrapper place-card__image-wrapper">
-          <a href="#">
+          <Link to={`${OFFER_URL}${id}`}>
```

Comment:
Следует использовать generatePath из реакт роутера
```suggestion
          <Link to={generatePath(AppRoute.Offer, { id: `${id}` )}>
```